x/sequencer/keeper/msg_server_bond.go

```diff
@@ -76,6 +76,9 @@ func (k msgServer) Unbond(goCtx context.Context, msg *types.MsgUnbond) (*types.M
 	// now we know they are proposer
 	// avoid starting another notice unnecessarily
 	if k.IsProposer(ctx, seq) {
+		if !k.rollappKeeper.ForkLatestAllowed(ctx, seq.RollappId) {
+			return nil, gerrc.ErrFailedPrecondition.Wrap("rotation could cause fork before genesis transfer")
+		}
```

Comment:
eagerly prevent rotation when it would end up in a fork before genesis transfer
better ux and prevents rollapp getting completely broken

Reply:
why u check it on the notice period **start**?
I think it will male more sense, to check on rotation start (notice period ends), in case nextProposer is sentinel